spec.md

```diff
@@ -15,4 +16,574 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
+| Workload          | The atomic unit of "work" scheduled by a COSI System. This MAY be a container or a collection of containers.                   |
+
+## Objective
+
+To define a standard “Container Object Storage Interface” (COSI) that will enable storage vendors (SP) to develop a plugin once and have it work across multiple Kubernetes Clusters and possibly on other COs.
+
+### Goals in MVP
+
+The Container Object Storage Interface (COSI) will:
+
+* Enable SP authors to write one COSI compliant Plugin that “just works” across all Kubernetes based clusters that implement COSI.
+* Define API (RPCs) that enable:
+  * Dynamic provisioning and de-provisioning of a bucket.
+  * Ability to define bucket access control mechanisms
+  * Consumption of both existing and new buckets created on SP.
+  * Define plugin protocol RECOMMENDATIONS.
+  * Describe a process by which a admin configures a Plugin.
+  * COSI deployments and templates. 
+
+### Non-Goals in MVP
+
+The Container Object Storage Interface (COSI) explicitly will not define, provide, or dictate:
+
+* Specific mechanisms by which a Plugin Supervisor manages the lifecycle of a Plugin, including:
+  * How to deploy, install, upgrade, uninstall, monitor, or respawn (in case of unexpected termination) Plugins.
+* A first class message structure/field to represent "grades of object storage" (aka "bucket class").
+* Protocol-level authentication and authorization.
+* Packaging of a Plugin.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and packaging recommendations for a storage provider (SP) to implement a COSI compatible plugin.
+The interface declares the RPCs that a plugin MUST expose: this is the **primary focus** of the COSI specification.
+Any operational and packaging recommendations offer additional guidance to promote cross-COSI system compatibility.
+
+### Architecture
+
+The primary focus of this specification is on the **protocol** between a COSI System and a Plugin.
+It SHOULD be possible to ship cross-COSI system compatible Plugins for a variety of deployment architectures.
+A COSI system SHOULD be equipped to handle both centralized and headless plugins.
+Several of these possibilities are illustrated in the following figures.
+
+```
+                    COSI System "Master" Host
++-------------------------------------------+
+|                                           |
+|  +------------+           +------------+  |
+|  |   COSI     |   gRPC    | Controller |  |
+|  |   System   +----------->   Plugin   |  |
+|  +------------+           +------------+  |
+|                                           |
++-------------------------------------------+
+
+```
+Figure 1: Plugin deployment, all of the COSI system Node hosts run Plugins. 
+
+
+![Cosi Components](cosicomponents.png)
+Figure 2: Interaction within COSI components 
```

Comment:
The diagram seems to be missing the border on top.

Reply:
Figure 2 assumes our Kubernetes implementation (eg. BucketRequest) so perhaps the caption should read something like:
`Figure 2: possible Kubernetes implementation showing Interaction of COSI components`

Reply:
fixed